File: app/src/main/java/com/whackyapps/pallavgrover/daresaydigital/activity/MainActivity.java
package com.whackyapps.pallavgrover.daresaydigital.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.whackyapps.pallavgrover.daresaydigital.R;
import com.whackyapps.pallavgrover.daresaydigital.fragment.FirstFragment;

public class MainActivity extends AppCompatActivity {

    private FirstFragment mFragment;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        mFragment = (FirstFragment) getSupportFragmentManager().findFragmentByTag(FirstFragment.TAG);
        if(mFragment == null) {
            mFragment = FirstFragment.newInstance();
            getSupportFragmentManager().beginTransaction().add(R.id.fragment_container, mFragment, FirstFragment.TAG).commit();
        }
    }

}

File: app/src/main/java/com/mousavi/hashem/mymoviesapp/presentaion/MainActivity.kt
package com.mousavi.hashem.mymoviesapp.presentaion

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.mousavi.hashem.mymoviesapp.R
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController
    private lateinit var bottomNavigationView: BottomNavigationView


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupBottomNavigation()

    }

    private fun setupBottomNavigation() {
        val navHostFragment = supportFragmentManager.findFragmentById(
            R.id.main_nav_host
        ) as NavHostFragment
        navController = navHostFragment.navController

        bottomNavigationView = findViewById(R.id.bottom_navigation_view)
        bottomNavigationView.setupWithNavController(navController)

        bottomNavigationView.setOnItemReselectedListener {
            if ((navController.currentDestination?.id == R.id.detailsFragment) ||
                navController.currentDestination?.id == R.id.reviewsFragment
            ) {
                navController.popBackStack()
            }
        }
    }

    fun selectExplorePage() {
        bottomNavigationView.selectedItemId = R.id.explore
    }

}